feat(tracing): Add `enableTracing` option

mydea opened the pull request:
This adds a new `enableTracing` option, which can be used to streamline performance collection.

* Setting this to `true` will default the `tracesSampleRate` basically to 1
* It is overwritten by `tracesSampleRate` and `tracesSampler`, if they are defined

This change also moves the `hasTracingEnabled` util to `@sentry/core`, so we can use it everywhere in a consistent way.

Closes https://github.com/getsentry/sentry-javascript/issues/7065

HazAT commented on packages/types/src/options.ts:
```diff
@@ -96,6 +96,13 @@ export interface ClientOptions<TO extends BaseTransportOptions = BaseTransportOp
 
+  /**
+   * If this is enabled, transactions and trace data will be generated and captured.
+   * This will set the `tracesSampleRate` to the recommended default of `1.0`.
```
```suggestion
   * This will set the `tracesSampleRate` to the recommended default of `1.0` if `tracesSampleRate` is undefined.
```

HazAT replied:
nit

AbhiPrasad reviewed (changes requested):
There's a couple other places in the codebase we also need to address.

https://github.com/getsentry/sentry-javascript/blob/f5b7b1013ee2d809944317b0fddf0da35f06caa6/packages/vue/src/sdk.ts#L83

https://github.com/getsentry/sentry-javascript/blob/f5b7b1013ee2d809944317b0fddf0da35f06caa6/packages/node/src/handlers.ts#L51

mydea commented:
So I moved the `hasTracingEnabled` util to `@sentry/core`, and deprecated the old export from `@sentry/tracing`.
I added a check for `__SENTRY_TRACING__` that should hopefully (?) lead to this function being replaced with `return false` when this is disabled.

mydea commented on packages/core/src/utils/hasTracingEnabled.ts:
```diff
@@ -0,0 +1,23 @@
+export function hasTracingEnabled(
+  maybeOptions?: Pick<Options, 'tracesSampleRate' | 'tracesSampler' | 'enableTracing'> | undefined,
+): boolean {
+  if (typeof __SENTRY_TRACING__ === 'boolean' && !__SENTRY_TRACING__) {
```
Hmm, I wonder if this is worth it 🤔 WDYT?

mydea replied:
I think the "downside" is that users that _do not_ configure this will/may still have this code in their bundle, right?

Lms24 reviewed (approved):
LGTM!

mydea commented:
@AbhiPrasad is this good by you now as well?

AbhiPrasad reviewed (approved):
🚀
